### PR TITLE
Prevent a possible segmentation fault (#4141)

### PR DESCRIPTION
**Pull request by jonipol**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #4141|
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo simulation of a Robauta robot |
| Does this PR contain AI generated software? | No |

---

## Description of contribution in a few bullet points
- Prevent a possible segmentation fault in analytic_expansion at smac_planner

Defensive approach as Steve recommended in his comment on the issue. Added check if the iterator == [ vector.end(), as dereferencing ](https://en.cppreference.com/w/cpp/container/vector/end) that can lead into undefined behavior (appeared as segmentation fault in my case). Also added an early return for the case in which the vector would end up being empty, thus leading into the check previously mentioned.

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_smac_planner/src/analytic_expansion.cpp [hunk 1]
```diff
@@ -189,6 +189,10 @@ typename AnalyticExpansion<NodeT>::AnalyticExpansionNodes AnalyticExpansion<Node
 
   // A move of sqrt(2) is guaranteed to be in a new cell
   static const float sqrt_2 = std::sqrt(2.0f);
+  if (d < sqrt_2) {
```
Comment (SteveMacenski) [inline nav2_smac_planner/src/analytic_expansion.cpp:192]: Add it to the `if` statement above 

```
if (d > _search_info.analytic_expansion_max_length || d < sqrt_2) {
```

Comment (SteveMacenski) [reply nav2_smac_planner/src/analytic_expansion.cpp:192]: ^ that would require moving the sqrt_2 definition above as well

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
LGTM, just a small detail for branching

File: nav2_smac_planner/src/analytic_expansion.cpp (hunk 1 above)
Comment (jonipol) [reply nav2_smac_planner/src/analytic_expansion.cpp]: Thanks for feedback!

**Comment by SteveMacenski** [comment]
Mergify backporting failed, will need to be done manually

**Pull request merged**